src/main/java/nl/delftelectronics/spaceinvaders/core/entities/Barricade.java

```diff
@@ -9,6 +11,8 @@
 public class Barricade extends SpriteEntity implements Collidable {
     private static final String FILENAME = "/barricade.png";
 
+    private EventListenerList destroyedEvent;
```

Comment:
Barricade is an Entity and thus already has a destroyedEvent, you can subscribe to that instead of implementing it again.